### DISCUSSION TITLE
在hooks-useEffect里面直接使用taro.redirectTo或者navigateTo跳转不生效

### DISCUSSION THREAD

**Discussion by 313973579** [NervJS/taro#10343] [Q & A]
![I`%@WDY @)R2K6$OO}3IV1C](https://user-images.githubusercontent.com/40569044/134790704-1870f7ca-fe8a-40b5-9169-0d8b895d7c59.jpg)

这样为什么地址栏上改成了index但是内容还是当前页的内容啊

但是如果使用setTimeout 500毫秒之后就可以

**Comment by Chen-jj** [comment]
H5 吗？

**Reply by 313973579** [reply]
是的

**Comment by Chen-jj** [comment]
@313973579 当时为了修别的问题，写死了两次跳转之间要间隔 500ms。。https://github.com/NervJS/taro/pull/10004

先延迟一下吧，这里改起来很麻烦，我们之后会重构一下
